### PR TITLE
returns an error if tx.Get_Node_By_Id returns nil node

### PR DESCRIPTION
**Pull request by navillasa**
## why changes
- The satellite was crashing with a nil pointer dereference panic whenever `getNodeStats(nodeID, dbNode)` was called where dbNode was nil. There needed to be a check that dbNode is not nil before passing it to `getNodeStats`.
- `getNodeStats` did not use its first parameter `nodeID` so it was removed.

## what changes
- Adds a check for if dbNode is nil and returns an error if so. However, **this isn't a great fix** and we definitely need to figure out why it's returning nil, since the docs say:
```
QueryRow executes a query that is expected to return at most one row.
QueryRow always returns a non-nil value.
```
(https://golang.org/pkg/database/sql/#DB.QueryRow). I'm going to create a sticky note for this.
- Removes `nodeID` from `getNodeStats` parameters.

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

**Comment by navillasa** [comment]
Stack trace from the original error:
```
2019-04-08T16:15:30.503Z        DEBUG   kademlia:endpoint       connection to node failed       {"error": "node error: transport error: connection error: desc = \"transport: error while dialing: dial tcp 18.216.53.253:13578: connect: connection refused\"", "errorVerbose": "node error: transport error: connection error: desc = \"transport: error while dialing: dial tcp 18.216.53.253:13578: connect: connection refused\"\n\tstorj.io/storj/pkg/transport.(*Transport).DialNode:88\n\tstorj.io/storj/pkg/kademlia.(*Dialer).dialNode:152\n\tstorj.io/storj/pkg/kademlia.(*Dialer).PingNode:81\n\tstorj.io/storj/pkg/kademlia.(*Kademlia).Ping:224\n
\tstorj.io/storj/pkg/kademlia.(*Endpoint).pingback:54\n\tstorj.io/storj/pkg/kademlia.(*Endpoint).Query:41\n\tstorj.io/storj/pkg/pb._Nodes_Query_Handler.func1:525\n\tstorj.io/storj/pkg/server.combineInterceptors.func1.1.1:50\n\tstorj.io/storj/pkg/auth/grpcauth.NewAPIKeyInterceptor.func1:27\n\tstorj.io/storj/pkg/server.combineInterceptors.func1.1:49\n\tstorj.io/storj/pkg/server.unaryInterceptor:35\n\tstorj.io/storj/pkg/server.combineInterceptors.func1:48\n\tstorj.io/storj/pkg/pb._Nodes_Query_Handler:527\n\tgoogle.golang.org/grpc.(*Server).processUnaryRPC:971\n\tgoogle.golang.org/grpc.(*Server).handleStream:1250\n\tgoogle.golang.org/g
rpc.(*Server).serveStreams.func1.1:690", "nodeID": "12EhPtzXgVK7inFCvapNhiXgNGKRX7vwGvf5WdNRYRYmH6a8rN3"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0xcce641]

goroutine 66 [running]:
gopkg.in/spacemonkeygo/monkit%2ev2.newSpan.func1(0xc000288780)
        /go/pkg/mod/gopkg.in/spacemonkeygo/monkit.v2@v2.0.0-20180827161543-6ebf5a752f9b/ctx17.go:150 +0x36a
panic(0xe8a7c0, 0x19e0200)
        /usr/local/go/src/runtime/panic.go:513 +0x1b9
gopkg.in/spacemonkeygo/monkit%2ev2.newSpan.func1(0xc0002887a0)
        /go/pkg/mod/gopkg.in/spacemonkeygo/monkit.v2@v2.0.0-20180827161543-6ebf5a752f9b/ctx17.go:150 +0x36a
panic(0xe8a7c0, 0x19e0200)
        /usr/local/go/src/runtime/panic.go:513 +0x1b9
storj.io/storj/satellite/satellitedb.getNodeStats(0x5b928eaf10e0be40, 0x83066303e7bb138d, 0x1d53f6a7b5fa1a73, 0x5ef1fc91a6756d, 0x0, 0x20)
        /go/storj.io/storj/satellite/satellitedb/overlaycache.go:733 +0x81
storj.io/storj/satellite/satellitedb.(*overlaycache).UpdateOperator(0xc00000ee48, 0x10ecdc0, 0xc0014b0460, 0x5b928eaf10e0be40, 0x83066303e7bb138d, 0x1d53f6a7b5fa1a73, 0x5ef1fc91a6756d, 0x0, 0x0, 0xc002b116b0, ...)
        /go/storj.io/storj/satellite/satellitedb/overlaycache.go:581 +0x438
storj.io/storj/pkg/overlay.(*Cache).UpdateOperator(0xc0002cf2c0, 0x10ec3c0, 0xc000303200, 0x5b928eaf10e0be40, 0x83066303e7bb138d, 0x1d53f6a7b5fa1a73, 0x5ef1fc91a6756d, 0x0, 0x0, 0xc002b116b0, ...)
        /go/storj.io/storj/pkg/overlay/cache.go:350 +0x18e
storj.io/storj/pkg/discovery.(*Discovery).refresh(0xc0000ba300, 0x10ec3c0, 0xc000303200, 0x3, 0x0)
        /go/storj.io/storj/pkg/discovery/service.go:163 +0xcf4
storj.io/storj/pkg/discovery.(*Discovery).Run.func1(0x10ec3c0, 0xc000303200, 0x3, 0x2)
        /go/storj.io/storj/pkg/discovery/service.go:82 +0x5e
storj.io/storj/internal/sync2.(*Cycle).Run(0xc0000ba328, 0x10ec3c0, 0xc000303200, 0xc0003f6530, 0x0, 0x0)
        /go/storj.io/storj/internal/sync2/cycle.go:118 +0x4ed
storj.io/storj/internal/sync2.(*Cycle).Start.func1(0x8, 0x100da70)
        /go/storj.io/storj/internal/sync2/cycle.go:58 +0x45
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0003fad50, 0xc0003fad80)
        /go/pkg/mod/golang.org/x/sync@v0.0.0-20190227155943-e225da77a7e6/errgroup/errgroup.go:57 +0x57
created by golang.org/x/sync/errgroup.(*Group).Go
        /go/pkg/mod/golang.org/x/sync@v0.0.0-20190227155943-e225da77a7e6/errgroup/errgroup.go:54 +0x66
kubectl --context prod -n mars-satellite logs -f mars-satellite-79f9b795cd-l7zh5: 05:15
```

File: satellite/satellitedb/overlaycache.go [hunk 1]
```diff
@@ -329,7 +329,14 @@ func (cache *overlaycache) CreateStats(ctx context.Context, nodeID storj.NodeID,
 		}
 	}
 
-	return getNodeStats(nodeID, dbNode), Error.Wrap(tx.Commit())
+	// TODO: Allegedly tx.Get_Node_By_Id and tx.Update_Node_By_Id should never return a nil value for dbNode,
```
Comment (ifraixedes) [inline satellite/satellitedb/overlaycache.go]: The implementation of [`tx.Update_Node_By_Id` returns nil when `QueryRow` returns `sql.ErrNoRows`](https://github.com/storj/storj/blob/master/satellite/satellitedb/dbx/satellitedb.dbx.go#L5754) so it looks an expected behavior for DBX update generated queries.

File: satellite/satellitedb/overlaycache.go [hunk 2]
```diff
@@ -329,7 +329,14 @@ func (cache *overlaycache) CreateStats(ctx context.Context, nodeID storj.NodeID,
...
+	// however we've seen from some crashes that it does. We need to track down the cause of these crashes
+	// but for now we're adding a nil check to prevent a panic.
+	if dbNode == nil {
+		return nil, Error.Wrap(errs.New("unable to get node by ID: %s", nodeID.String()))
```
Comment (egonelbre) [inline satellite/satellitedb/overlaycache.go]: This leaks a transaction when something fails.

File: satellite/satellitedb/overlaycache.go [hunk 3]
```diff
@@ -442,8 +449,14 @@ func (cache *overlaycache) UpdateStats(ctx context.Context, updateReq *overlay.U
 		return nil, Error.Wrap(errs.Combine(err, tx.Rollback()))
 	}
 
-	nodeStats := getNodeStats(nodeID, dbNode)
-	return nodeStats, Error.Wrap(tx.Commit())
+	// TODO: Allegedly tx.Get_Node_By_Id and tx.Update_Node_By_Id should never return a nil value for dbNode,
+	// however we've seen from some crashes that it does. We need to track down the cause of these crashes
+	// but for now we're adding a nil check to prevent a panic.
+	if dbNode == nil {
+		return nil, Error.Wrap(errs.New("unable to get node by ID: %s", nodeID.String()))
```
Comment (egonelbre) [inline satellite/satellitedb/overlaycache.go]: This leaks a transaction when something fails.

File: satellite/satellitedb/overlaycache.go [hunk 4]
```diff
@@ -502,9 +515,14 @@ func (cache *overlaycache) UpdateUptime(ctx context.Context, nodeID storj.NodeID
 	if err != nil {
 		return nil, Error.Wrap(errs.Combine(err, tx.Rollback()))
 	}
+	// TODO: Allegedly tx.Get_Node_By_Id and tx.Update_Node_By_Id should never return a nil value for dbNode,
+	// however we've seen from some crashes that it does. We need to track down the cause of these crashes
+	// but for now we're adding a nil check to prevent a panic.
+	if dbNode == nil {
+		return nil, Error.Wrap(errs.New("unable to get node by ID: %s", nodeID.String()))
```
Comment (egonelbre) [inline satellite/satellitedb/overlaycache.go]: This leaks a transaction when something fails.

**Pull request merged**